server/controllers/userEventController.ts

```diff
@@ -0,0 +1,162 @@
+import { StatusCodes } from "http-status-codes";
+import { Request, Response } from "express";
+import { DatabaseUserEvent } from "../services/database.user.event";
+
+import UnauthorizedError from "../errors/UnauthorizedError";
+import mongoose, { ObjectId } from "mongoose";
+import BadRequestError from "../errors/BadRequest";
+import { UserEventResponse } from "../types/responses/userEventResponse";
+import {
+	isValidPartialUserEventRequest,
+	isValidUserEventRequest
+} from "../utils/isValidRequest";
+import { includeIfDefined } from "../utils/includeIfDefined";
+
+export const create = async (req: Request, res: Response) => {
+	if (!req.session.userid) {
+		throw new UnauthorizedError();
+	}
+
+	const data: unknown = req.body;
+	if (!isValidUserEventRequest(data)) {
+		throw new BadRequestError("Invalid UserEvent Request data");
+	}
+
+	try {
+		const event: UserEventResponse = await DatabaseUserEvent.create(data);
+
+		res.status(StatusCodes.CREATED).send({
+			event,
+			message: "Event created"
+		});
+	} catch (error) {
+		throw new Error();
+	}
+};
+
+interface ReadQueryParam {
+	name?: string;
+}
+
+export const read = async (req: Request, res: Response) => {
+	const userid: undefined | ObjectId = req.session.userid;
+	if (!userid) {
+		throw new UnauthorizedError();
+	}
+
+	const { name }: ReadQueryParam = req.query;
+	if (!name) {
+		throw new BadRequestError();
+	}
+
+	const event: UserEventResponse | undefined = await DatabaseUserEvent.read(
+		userid,
+		name
+	);
+
+	let statusCode: StatusCodes;
+	let message: string;
+
+	if (!event) {
+		statusCode = StatusCodes.NOT_FOUND;
+		message = "No events found";
+	} else {
+		statusCode = StatusCodes.OK;
+		message = "Events found";
+	}
+
+	res.status(statusCode).send({
+		...includeIfDefined("event", event),
+		message
+	});
+};
+
+interface ListQueryParams {
+	day?: string;
+	month?: string;
+	year?: string;
+}
+
+export const listEvents = async (req: Request, res: Response) => {
+	const userid: undefined | ObjectId = req.session.userid;
+	if (!userid) {
+		throw new UnauthorizedError();
+	}
+
+	const { day, month, year }: ListQueryParams = req.query;
+	if (!day || !month || !year) {
+		throw new BadRequestError("Missing query params");
+	}
+
+	const events: UserEventResponse[] | undefined =
+		await DatabaseUserEvent.listEvents(
+			userid,
+			parseInt(day),
+			parseInt(month),
+			parseInt(year)
+		);
+
+	let statusCode: StatusCodes;
+	let userEvents: UserEventResponse[];
+	let message: string;
+
+	if (!events) {
+		userEvents = [];
+		statusCode = StatusCodes.NOT_FOUND;
```

Comment:
not sure if a 404 is good to send here. It implies that the endpoint does not exist. However, the endpoint does exist. So if the query returns nothing send 200. if the query throws error we should send a 500